.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,27 @@
+name: Pre-commit
+
+on: [ pull_request ]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: Install terraform-docs
```

Comment:
I would recommend using pre-commit as it's intended for, on a local machine. We expect engineers to contribute (push) code that's conform our standards to the upstream.

Terraform-docs, validate, fmt and checkov should be ran locally using pre-commit.

Whereas checkov as a check can be ran as part of a pull request.